packages/client/lib/AccessTokenClient/IssuanceInitiationAccessTokenClient.ts

```diff
@@ -13,38 +13,55 @@ import {
 import { ObjectUtils } from '@sphereon/ssi-types';
 import Debug from 'debug';
 
-import { MetadataClient } from './MetadataClient';
-import { convertJsonToURI, formPost } from './functions';
+import { IssuanceInitiationClient } from '../CredentialOffer';
+import { MetadataClient } from '../MetadataClient';
+import { convertJsonToURI, formPost } from '../functions';
+
+import { AccessTokenClient } from './AccessTokenClient';
 
 const debug = Debug('sphereon:openid4vci:token');
 
-export class AccessTokenClient {
-  public async acquireAccessTokenUsingIssuanceInitiation({
-    issuanceInitiation,
-    asOpts,
-    pin,
-    codeVerifier,
-    code,
-    redirectUri,
-    metadata,
-  }: AccessTokenRequestOpts): Promise<OpenIDResponse<AccessTokenResponse>> {
-    const { issuanceInitiationRequest } = issuanceInitiation;
+export class IssuanceInitiationAccessTokenClient implements AccessTokenClient {
+  getAccessTokenRequest = (
+    issuanceInitiationClient: IssuanceInitiationClient,
+    metadata: EndpointMetadata,
+    pin?: string,
+    clientId?: string,
+    codeVerifier?: string,
+    code?: string,
+    redirectUri?: string
+  ): IssuanceInitiationAccessTokenRequestOpts => {
+    return {
+      issuanceInitiation: issuanceInitiationClient.issuanceInitiationWithBaseUrl,
```

Comment:
Please just stick to the assignment at hand. The previous destructured aproach was more legable backed by an interface

Reply:
reverted